src/perf_event/perf_event_array.rs

```diff
@@ -0,0 +1,87 @@
+use crate::core::BPF;
+use crate::error::BccError;
+use crate::perf_event::{Event, EventType};
+
+#[derive(Default)]
+pub struct PerfEventArray {
+    table: Option<String>,
+    event: Option<Event>,
+    cpu: Option<usize>,
+}
+
+/// A `PerfEventArray` is used to configure a BPF `PERF_EVENT_ARRAY` such that a call
+/// `table.perf_read(CUR_CPU_IDENTIFIER) returns the hardware counter of the event
+/// on the local cpu.
+impl PerfEventArray {
+    /// Creates a new `PerfEventArray` with defaults.There are several mandatory
+    /// fields which must be configured before attaching.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Specify the name of the table within the BPF code. This is a
+    /// required item.
+    pub fn table(mut self, name: &str) -> Self {
+        self.table = Some(name.to_owned());
+        self
+    }
+
+    /// The `Event` which will cause a probe to fire, such as a hardware or
+    /// software event. This is required.
+    pub fn event(mut self, event: Event) -> Self {
+        self.event = Some(event);
+        self
+    }
+
+    /// Restrict the probe to only the given hardware thread. If this is set to
+    /// `None` (the default), a probe will be created for each hardware thread
+    /// on the system to provide system-wide coverage.
+    pub fn cpu(mut self, cpu: Option<usize>) -> Self {
+        self.cpu = cpu;
+        self
+    }
+
+    /// Consumes the perf event and opens it. May return an error if there is a
+    /// incomplete or invalid configuration or other error while loading or
+    /// opening the event.
+    pub fn open(self, bpf: &mut BPF) -> Result<(), BccError> {
```

Comment:
I'd like to keep the terminology consistent across all of these builder-esque types so it's less burden for users. Can we rename to `attach` to match?